app/api/route/user_route.py

```diff
@@ -22,21 +22,22 @@
 @user_blueprint.input(user_schemas.UserSchema)
 @user_blueprint.output(user_schemas.UserSchema, status_code=201)
 @user_blueprint.auth_required(api_key_auth)
-def user_post(user_params: users.CreateUserParams) -> dict:
+@flask_db.with_db_session
+def user_post(db_session: db.Session, user_params: users.CreateUserParams) -> dict:
     """
     POST /v1/user
     """
     logger.info("POST /v1/user")
 
-    with flask_db.get_db(current_app).get_session() as db_session:
-        user = user_service.create_user(db_session, user_params)
+    user = user_service.create_user(db_session, user_params)
 
-        logger.info(
-            "Successfully inserted user",
-            extra=get_user_log_params(user),
-        )
+    logger.info(
+        "Successfully inserted user",
+        extra=get_user_log_params(user),
+    )
```

Comment:
I realized we still want to start the within the service method so that the service method can control how it completes it's work of flushing to the db. in general, when transactions should start/stop is determined by business logic, so the service should be responsible for that, not the framework.

i noticed that in this case when we started the transaction in the decorator, the create_user method hasn't completed it's work by this line of code, so get_user_log_params still shows user_id as being None

Reply:
You can work around that by doing a flush, or generating the uuid and setting it on the object yourself, but I think being explicit with transactions makes sense. Technically all of the Mass PFML code outside a few explicit commits did it exactly like that, but it never seemed to cause any issues.

Somewhat related, would you find it useful to be able to "register" parameters during a request (like the user ID here) and have them automatically attached to any subsequent logs? I added a utility like that in NJ that expands on https://github.com/navapbc/template-application-flask/blob/main/app/api/logging/log_attributes.py

Reply:
Yeah I'd love to standardize on how we do logging to optimize for observable and debuggable code. Registering parameters sounds good, could you share the approach you did? Easy to adapt / port over?

Also, looks like [subclasses of Logger can override makeRecord](https://docs.python.org/3/library/logging.html#logging.Logger.makeRecord). I wonder if that could be of any use.

Reply:
I added this function to log_attributes:

```py

def add_request_log_attribute(key: str, value: Any) -> None:
    """
    For he duration of the current flask request, add
    a parameter that will appear in the extra section
    of any log statement. Useful for attaching an ID
    of a record relevant to a request.
    """

    if flask.has_request_context():
        request_log_attributes: dict[str, Any] = flask.g.get("request_log_attributes", {})
        request_log_attributes[key] = value

        flask.g.request_log_attributes = request_log_attributes
```

And added the following to `get_request_attributes`:

```py
        request_log_attributes = flask.g.get("request_log_attributes")
        if request_log_attributes:
            request_attributes |= request_log_attributes
```

Basically just continuing the pattern. It does use the flask `g` context, but since that's tied to the request context, I think it makes sense to keep using it like this since it's log info only relevant to the given request.

Reply:
Oh I see! Neat, I'll give it a think 👍 
in the meantime i created this placeholder ticket for that idea https://github.com/navapbc/template-application-flask/issues/87